src/lib.rs

```diff
@@ -15,18 +15,58 @@ pub enum CharacterSet {
     UrlSafe
 }
 
+#[derive(Clone, Copy, Debug)]
+pub enum LineEnding {
+    LF,
+    CRLF,
+}
+
 /// Contains configuration parameters for base64 encoding
 #[derive(Clone, Copy, Debug)]
 pub struct Config {
     /// Character set to use
     pub char_set: CharacterSet,
     /// True to pad output with `=` characters
     pub pad: bool,
+    /// Remove whitespace before decoding, at the cost of an allocation
+    pub strip_whitespace: bool,
+    /// Characters per line, None (or Some(0), but less "proper") for no linebreaks
+    pub line_size: Option<usize>,
+    /// Unix or Windows line endings, ignored if above None/Some(0)
+    pub line_ending: LineEnding,
```

Comment:
was actually going to change to
```
pub enum LineWrap {
    NoWrap, 
    Wrap(usize, LineEnding)
}
```
to be more succinct, but a private struct could work too (also an ADT doesn't solve the potential confusion with 0, tfw no dependent types)

tbh I think just having the one config struct is fine for now. I'd honestly be surprised if many people even used it, I'd expect 90% of usecases to be encode/decode with implicit STANDARD and most of the rest using one of the other predefined ones